In [2]:
import pandas as pd

In [3]:
dataset = pd.read_csv("preprocessed_placement.csv")

In [4]:
dataset

ssc_p  hsc_p  degree_p  etest_p  mba_p         salary gender    ssc_b  \
0    67.00  91.00     58.00     55.0  58.80  270000.000000      M   Others   
1    79.33  78.33     77.48     86.5  66.28  200000.000000      M  Central   
2    65.00  68.00     64.00     75.0  57.80  250000.000000      M  Central   
3    56.00  52.00     52.00     66.0  59.43  288655.405405      M  Central   
4    85.80  73.60     73.30     96.8  55.50  425000.000000      M  Central   
..     ...    ...       ...      ...    ...            ...    ...      ...   
210  80.60  82.00     77.60     91.0  74.49  400000.000000      M   Others   
211  58.00  60.00     72.00     74.0  53.62  275000.000000      M   Others   
212  67.00  67.00     73.00     59.0  69.72  295000.000000      M   Others   
213  74.00  66.00     58.00     70.0  60.23  204000.000000      F   Others   
214  62.00  58.00     53.00     89.0  60.22  288655.405405      M  Central   

       hsc_b     hsc_s   degree_t workex specialisation      status  
0     Others  Commerce   Sci&Tech     No         Mkt&HR      Placed  
1     Others   Science   Sci&Tech    Yes        Mkt&Fin      Placed  
2    Central      Arts  Comm&Mgmt     No        Mkt&Fin      Placed  
3    Central   Science   Sci&Tech     No         Mkt&HR  Not Placed  
4    Central  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
..       ...       ...        ...    ...            ...         ...  
210   Others  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
211   Others   Science   Sci&Tech     No        Mkt&Fin      Placed  
212   Others  Commerce  Comm&Mgmt    Yes        Mkt&Fin      Placed  
213   Others  Commerce  Comm&Mgmt     No         Mkt&HR      Placed  
214   Others   Science  Comm&Mgmt     No         Mkt&HR  Not Placed  

[215 rows x 14 columns]

In [5]:
dataset.isnull().sum()

ssc_p             0
hsc_p             0
degree_p          0
etest_p           0
mba_p             0
salary            0
gender            0
ssc_b             0
hsc_b             0
hsc_s             0
degree_t          0
workex            0
specialisation    0
status            0
dtype: int64

In [6]:
#covarience
dataset.cov(numeric_only=True)

ssc_p         hsc_p     degree_p        etest_p  \
ssc_p       117.228377     60.348373    42.897137      37.659225   
hsc_p        60.348373    118.755706    34.819820      35.461678   
degree_p     42.897137     34.819820    54.151103      21.929469   
etest_p      37.659225     35.461678    21.929469     176.251018   
mba_p        24.535952     22.555846    17.272020      16.886973   
salary    19767.643976  46008.057275 -8064.357161  157157.850783   

                 mba_p        salary  
ssc_p        24.535952  1.976764e+04  
hsc_p        22.555846  4.600806e+04  
degree_p     17.272020 -8.064357e+03  
etest_p      16.886973  1.571579e+05  
mba_p        34.028376  6.611551e+04  
salary    66115.509283  5.999726e+09

# Assignment I
# what is  the kind of Covarience between degress_p and etest_p marks
The covarience value of degree_p and etest_p is 21.929469 which is a postive values so it is positive covarience and both increase/decrease together with difference 21%
# what is  the kind of Covarience between etest_p and mba_p marks
The covarience value of etest_p and mba_p is 16.886973 which is a postive values so it is positive covarience and both are increase/decrease together with difference 16%

In [12]:
dataset.corr(numeric_only=True)

ssc_p     hsc_p  degree_p   etest_p     mba_p    salary
ssc_p     1.000000  0.511472  0.538404  0.261993  0.388478  0.023571
hsc_p     0.511472  1.000000  0.434206  0.245113  0.354823  0.054506
degree_p  0.538404  0.434206  1.000000  0.224470  0.402364 -0.014148
etest_p   0.261993  0.245113  0.224470  1.000000  0.218055  0.152829
mba_p     0.388478  0.354823  0.402364  0.218055  1.000000  0.146324
salary    0.023571  0.054506 -0.014148  0.152829  0.146324  1.000000

# Assignment II:
# what is the kind of Correlation between salary and mba_p marks
Correlation between mba_p and salary is 0.146324 which is less than 0.54 so it is low degree or weak positive correlation. both increase  with 14%